src/library/scala/beans/ScalaBeanInfo.scala

```diff
@@ -32,15 +31,9 @@ abstract class ScalaBeanInfo(clazz: java.lang.Class[_],
   override def getPropertyDescriptors() = pd
   override def getMethodDescriptors() = md
 
-  // override def getAdditionalBeanInfo() = Array(Introspector getBeanInfo clazz.getSuperclass)
-
   private def init() {
-    var i = 0
```

Comment:
It's true that three objects additional are created, but in relative terms compared to the `withFilter` object created a couple of lines above, along with the boxed `getMethods` array, the builder object, the resulting `md` collection with the size equal to the number of methods, and the array for `pd`... this may be insignificant.

Also - ranges should be optimized by the compiler anyway, and the whole `foreach` on the range should really be inlined and transformed into a `while` loop with the optimized build (not 100% sure, but I think this used to be the case).